### ISSUE TITLE
react client fails after upgrade

### ISSUE THREAD

**Issue by manusauvage** [jfurrow/flood#280]
Hi!

Somewhere between commit 4942ef9 and ca3fbba (I suspect commit 1904273 is the culprit), the react client in flood stopped working on my setup. The server part is still functionnal (rss are checked, torrents are queued), but the gui remains stuck with two items not loading (_torrent taxonomy_ and _data transfer rate details_).

Here's a dump of the browser console, in case it helps:
```
Unhandled promise rejection TypeError: r.download is undefined
Stack trace:
y</<.value@https://myserver/flood/app.js:36:20800
y</<.value@https://myserver/flood/app.js:36:22468
y</<.value@https://myserver/flood/app.js:36:19148
o</e.prototype.notifyAll@https://myserver/flood/app.js:8:21013
h.close@https://myserver/flood/app.js:10:28784
i.closeAll@https://myserver/flood/app.js:8:24426
i.perform@https://myserver/flood/app.js:8:23918
i.perform@https://myserver/flood/app.js:8:23838
.perform@https://myserver/flood/app.js:8:19762
k@https://myserver/flood/app.js:8:19939
i.closeAll@https://myserver/flood/app.js:8:24426
i.perform@https://myserver/flood/app.js:8:23918
f.batchedUpdates@https://myserver/flood/app.js:10:25923
u@https://myserver/flood/app.js:8:18974
r@https://myserver/flood/app.js:10:21047
l<.enqueueSetState@https://myserver/flood/app.js:10:22059
r.prototype.setState@https://myserver/flood/app.js:5:27436
H</<.value@https://myserver/flood/app.js:36:7707
n.prototype.emit@https://myserver/flood/app.js:14:16862
R</<.value@https://myserver/flood/app.js:36:12070
x.dispatcherID<@https://myserver/flood/app.js:36:12948
o</e.prototype._invokeCallback@https://myserver/flood/app.js:13:31316
o</e.prototype.dispatch@https://myserver/flood/app.js:13:31131
_</<.value@https://myserver/flood/app.js:13:30156
m.fetchTransferData/<@https://myserver/flood/app.js:15:2113
R/</o@https://myserver/flood/app.js:2:12037
R/<@https://myserver/flood/app.js:2:12159
e.exports/l@https://myserver/flood/app.js:2:16281
```

Flood is running on a Debian system using the upstream version of nodejs (v6.10.0), and all dependencies should be satisfied.

Could you lend me a hand here? How can I further help to solve the issue?
As a temporary fix I rolled back to 4942ef9, but I'd love it if I could stay up-to-date :)

Thanks,

**Comment by jfurrow** [comment]
Sorry you're having issues @manusauvage. Can you paste the output you see from the Node server when this happens?

**Comment by manusauvage** [comment]
As it appears, that bug was caused by the combination of:
- aggressive caching on the reverse proxy
- a config.js regression (somehow the baseURI value was reset to /)
- a global, catch all rule redirecting all failed requests to /, which served valid HTML content.

I still have no idea why reverting to 4942ef9 did the trick for a while, which led me to think this was a bug in the code. Anyway I started fresh again, fixed the config and everything's working like a charm.

Sorry for the noise, and thanks for the code :)
